File: Study-Up/app/src/main/java/com/ort/studyup/common/models/Deck.kt
package com.ort.studyup.common.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class Deck(
    id: Int,
    author: String,
    name: String,
    difficulty: Int,
    subject: String,
    isHidden: Boolean,
    val flashcards: List<Flashcard>,
) : DeckData(id, author, name, difficulty, subject, isHidden)

open class DeckData(
    @SerializedName("id")val id: Int = -1,
    @SerializedName("author")val author: String,
    @SerializedName("name")val name: String,
    @SerializedName("difficulty")val difficulty: Int,
    @SerializedName("subject")val subject: String,
    @SerializedName("isHidden")val isHidden: Boolean,
    @SerializedName("flashcardCount")val flashcardCount: Int? = null,
) : Serializable

class NewDeckRequest(
    @SerializedName("name") val name: String,
    @SerializedName("difficulty") val difficulty: Int,
    @SerializedName("subject") val subject: String,
    @SerializedName("isHidden") val isHidden: Boolean,
)

File: StudyUp/Domain/Deck.cs
using Domain.Enumerations;
using System.Collections.Generic;

namespace Domain
{
    public class Deck
    {
        public int Id { get; set; }
        public virtual User Author { get; set; }
        public virtual List<DeckGroup> DeckGroups { get; set; }
        public Difficulty Difficulty { get; set; }
        public virtual List<Flashcard> Flashcards { get; set; }
        public bool IsHidden { get; set; }
        public string Name { get; set; }
        public string Subject { get; set; }

        public Deck()
        {
            IsHidden = false;
        }

        public override bool Equals(object o)
        {
            if (o == null || !GetType().Equals(o.GetType()))
            {
                return false;
            }
            else
            {
                Deck anotherDeck = (Deck)o;
                return this.Name.ToString().Equals(anotherDeck.Name.ToString());
            }
        }
    }
}
